simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -51,8 +51,45 @@ void EntityBase::requestSpeedChange(
 {
   switch (transition) {
     case speed_change::Transition::LINEAR: {
-      setAccelerationLimit(std::fabs(constraint.value));
-      setDecelerationLimit(std::fabs(constraint.value));
+      if (getStatus().action_status.twist.linear.x < target_speed) {
+        setAccelerationLimit(std::fabs(constraint.value));
+        job_list_.append(
+          /**
+           * @brief Checking if the entity reaches target speed.
+           */
+          [this, target_speed]() {
+            if (getStatus().action_status.twist.linear.x >= target_speed) {
+              return true;
+            }
+            return false;
```

Comment:
This statement seems overly redundant.
If the return value is a bool value, it would be better to return the conditional expression of the "if" part as is to reduce the possibility of introducing errors.

```
return getStatus().action_status.twist.linear.x <= target_speed;
```

Reply:
fixed in https://github.com/tier4/scenario_simulator_v2/pull/752/commits/36eb578517a9090d6f526061baf16f63b7e6fe4c